serialize/src/lib.rs

```diff
@@ -212,7 +212,7 @@ pub fn buffer_bit_byte_size(modulus_bits: usize) -> (usize, usize) {
 /// into the number of bytes required to represent it.
 #[inline]
 pub const fn buffer_byte_size(modulus_bits: usize) -> usize {
-    (modulus_bits + 7) / 8
+    (modulus_bits + 1 + 7) / 8
```

Comment:
This isn’t necessarily correct; it adds a byte to every representation, no? Can we add a test to ensure that it maintains the old behaviour for smaller moduli?

and is the reason the test is failing because you’re trying to use the existing arithmetic infrastructure for 256 bit moduli?

Reply:
If you’re trying to use the existing infrastructure for secp256k1, I think a better idea would be to port over the algorithms from here: https://github.com/pornin/crrl/blob/3a3e625d784b4234d53829771fc3e18abab16913/src/backend/w64/modint.rs#L141

that way we introduce the extra limb only when necessary.

Reply:
Let me take a look at this new algorithm---it would be a cleaner fix and is also for the future.